tsdb/shard_mapper.go

```diff
@@ -0,0 +1,434 @@
+package tsdb
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/influxdb/influxdb/influxql"
+)
+
+type ShardMapper interface {
+	// Open will open the necessary resources to being the map job. Could be connections to remote servers or
+	// hitting the local bolt store
+	Open() error
+
+	// Close will close the mapper (either the bolt transaction or the request)
+	Close()
+
+	// Begin will set up the mapper to run the map function for a given aggregate call starting at the passed in time.
+	Begin(aggregate *influxql.Call, startingTime int64, limit int) error
```

Comment:
This should have an interface like this:

```go
// the start time should be part of the statement
Begin(stmt *influxql.Statement, chunkSize int) error
```

I call it `chunkSize` because that's what this number is. `limit` is already used elsewhere in the query language so it's best to steer clear of that since it could be confusing as these are different concepts. Chunks are for limiting the number of values buffered in memory before flushing out to the coordinator node.